tools/dh_eoscontent

```diff
@@ -111,20 +111,30 @@ sub app_appdata_file {
 	my $tmp=shift;
 
 	# See if this package has the appdata file
-	my $appdata = $tmp . '/usr/share/appdata/' . $app_id
-	    . '.appdata.xml';
-	verbose_print("Checking for app-data file $appdata");
-	if (-l $appdata) {
-		my $link = readlink($appdata);
-		if ($link =~ /^\//) {
-			# If the symlink is absolute, make it relative
-			# to the tmpdir
-			$appdata = $tmp . $link;
-		} else {
-			# Resolve it to an absolute path
-			$appdata = abs_path($appdata);
+	my $appdata;
+	my @appdata_dirs = ('metainfo', 'appdata');
+	foreach my $appdata_dir (@{appdata_dirs}) {
+		my $appdata_file = $tmp . '/usr/share/' . $appdata_dir . '/' . $app_id
+		    . '.appdata.xml';
+		verbose_print("Checking for app-data file $appdata_file");
+
+		next unless (-e $appdata_file);
+		$appdata = $appdata_file;
```

Comment:
I don't think the extra `appdata_file` variable is needed. I think you could assign directly to `$appdata` since nothing else is done with it besides checking it's existence.

Reply:
I think you're right, I added it just to be explicit on that `$appdata` would only be set if the file exists on disk. But you're right on that if that's not the case we would simply exhaust the 2 iterations of the loop and check anyway for its existence before returning, so I'll change that.